packages/wallet-adapter-core/src/WalletCore.ts

```diff
@@ -307,4 +310,36 @@ export class WalletCore extends EventEmitter<WalletCoreEvents> {
       throw new WalletNetworkChangeError(errMsg).message;
     }
   }
+
+  async signMessageAndVerify(message: SignMessagePayload): Promise<boolean> {
+    try {
+      this.isWalletExists();
+      if (!this._account) throw new Error("No account found!");
+      const response = await this._wallet?.signMessage(message);
+      if (!response)
+        throw new WalletSignMessageAndVerifyError("Failed to sign a message")
+          .message;
+      // Verify that the bytes were signed using the private key that matches the known public key
+      let verified = false;
+      // Remove the 0x prefix
+      const currentAccountPublicKey = this._account.publicKey.slice(2, 66);
```

Comment:
What if they don't have hex prefix?

Reply:
I think there's a verify method on AccountAddress in Aptos package

Reply:
thanks! I think there is a WIP validate method to add to the ts sdk, but was able to find a way to ensure this is a hex string and then remove its prefix to use it in the signature.verify method